### PR TITLE
Release/1.5.0

### PR DESCRIPTION
**Pull request by ThinkTankShark**
- Adds isEqualArray to Utils to shallow compare 2 arrays

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -252,3 +252,16 @@ export function detectIEVersion() {
...
+
+/**
+ * Shallow comparison between 2 arrays
+ * @param  {Array} arr1
+ * @param  {Array} arr2
+ * @return {Bool}
+ */
+export function isEqualArray(arr1, arr2) {
+  return (
+    arr1.length === arr2.length &&
```
Comment (isuttell) [inline src/utils.js]: I might add a check to ensure arr1 and arr2 are both arrays to begin

File: src/utils.js [hunk 2]
```diff
@@ -252,3 +252,16 @@ export function detectIEVersion() {
...
+    arr1.every((element, index) => element === arr2[index])
```
Comment (isuttell) [inline src/utils.js]: From a design standpoint we might want to add the option to add a custom comparator so if you want to check to do more advanced or specific checks you can.

Comment (ThinkTankShark) [reply src/utils.js]: Sure added.

File: src/utils.js (hunk 1 above)
Comment (ThinkTankShark) [reply src/utils.js]: Sure added

File: src/utils.js [hunk 3]
```diff
@@ -257,9 +257,20 @@ export function detectIEVersion() {
...
-export function isEqualArray(arr1, arr2) {
+export function isEqualArray(arr1, arr2, cb) {
+  // return false if both are not an array
+  if (!Array.isArray(arr1) || !Array.isArray(arr2)) {
+    return false;
+  }
+
+  // envoke the callback function if there is any
+  if (typeof cb === 'function') {
+    return cb(arr1, arr2);
```
Comment (isuttell) [inline src/utils.js]: This wasn't quite what I was thinking. I was thinking you could pass in a custom comparator in the `every` method. So something like:

```
function compare(arr1, arr2, fn) {
  if (typeof fn !== 'function') {
    fn = (element, index) => elenent === arr2[index];
 }
 return arr1.every(fn);
}

```

Doing it that way means you don't have to duplicate any checks if you want to override the logic

**Pull request merged**